deals/types.go

```diff
@@ -31,6 +31,7 @@ type DealInfo struct {
 	Size     uint64
 
 	PricePerEpoch uint64
+	StartEpoch    uint64
```

Comment:
`StartEpoch` is at which epoch the Deal will start charging the client.
`ActivationEpoch` is at which epoch the deal could be considered active for retrieval.

Both are different things (Activation <= Start).

Reply:
Actually, many others were missing. Updated.